src/main/java/de/unistuttgart/chickenshockbackend/service/GameResultService.java

```diff
@@ -1,62 +1,87 @@
-                throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("There is no question with uuid %s.", roundResultDTO.getQuestionUUId()));
-            } else {
-                RoundResult roundResult = new RoundResult(questionToAdd.get(),roundResultDTO.getAnswer());
-                questionList.add(roundResult);
-             }
-        }
-        return questionList;
-    }
-
-    /**
-     * Casts a GameResultDTO to GameResult and saves it in the Database
-     *
-     * @param gameResultDTO extern gameResultDTO
-     */
-    public void saveGameResult(GameResultDTO gameResultDTO) {
-        List<RoundResult> correctQuestions = this.castQuestionList(gameResultDTO.getCorrectAnsweredQuestions());
-        List<RoundResult> wrongQuestions = this.castQuestionList(gameResultDTO.getWrongAnsweredQuestions());
-        //String playerId = authorizationService.getPlayerId(token);TODO: after login is implemented
-        GameResult result = new GameResult(gameResultDTO.getQuestionCount(), gameResultDTO.getTimeLimit(), gameResultDTO.getFinishedInSeconds(), gameResultDTO.getCorrectKillsCount(), gameResultDTO.getWrongKillsCount(), gameResultDTO.getKillsCount(), gameResultDTO.getShotCount(), gameResultDTO.getPoints(), correctQuestions, wrongQuestions, gameResultDTO.getConfigurationAsUUID(), "playerId");
-        gameResultRepository.save(result);
-    }
+  @Autowired
+  ResultClient resultClient;
+
+  @Autowired
+  GameResultRepository gameResultRepository;
+
+  @Autowired
+  QuestionRepository questionRepository;
+
+  @Autowired
+  AuthorizationService authorizationService;
+
+  @Autowired
+  RoundResultMapper roundResultMapper;
+
+  /**
+   * Casts a GameResultDTO to GameResult and saves it in the Database
+   *
+   * @param gameResultDTO extern gameResultDTO
+   */
+  public void saveGameResult(final GameResultDTO gameResultDTO) {
+    final List<RoundResult> correctQuestions = roundResultMapper.roundResultDTOsToRoundResults(gameResultDTO.getCorrectAnsweredQuestions());
+    final List<RoundResult> wrongQuestions = roundResultMapper.roundResultDTOsToRoundResults(gameResultDTO.getWrongAnsweredQuestions());
+    //String playerId = authorizationService.getPlayerId(token);TODO: after login is implemented
+    final GameResult result = new GameResult(
+      gameResultDTO.getQuestionCount(),
+      gameResultDTO.getTimeLimit(),
+      gameResultDTO.getFinishedInSeconds(),
+      gameResultDTO.getCorrectKillsCount(),
+      gameResultDTO.getWrongKillsCount(),
+      gameResultDTO.getKillsCount(),
+      gameResultDTO.getShotCount(),
+      gameResultDTO.getPoints(),
+      correctQuestions,
+      wrongQuestions,
+      gameResultDTO.getConfigurationAsUUID(),
+      "playerId"
+    );
+    gameResultRepository.save(result);
+
+    final int resultScore = calculateResultScore(
+      result.getCorrectKillsCount(),
+      result.getWrongKillsCount(),
+      result.getQuestionCount(),
+      result.getTimeLimit()
+    );
```

Comment:
Why the time limit has an impact on the score? This is no performance from the player?
Do you maybe mean `FinishedInSeconds`?

```suggestion
    final int resultScore = calculateResultScore(
      result.getCorrectKillsCount(),
      result.getWrongKillsCount(),
      result.getQuestionCount(),
      result.getFinishedInSeconds()
    );
```

Reply:
The idea is, that you only have a certain time per question to get a 100% score. The current time is 15 sec per question.

Reply:
Yes I got this. But currently you use the timelimit of the game and not the time the player needed to answer the questions

Reply:
you shouldn't be punished for beeing faster than the timelimit.

Reply:
Yes, but `FinishedInSeconds` is the time limit of the game and always the same, independent how long the user needs to answer the questions. 

```
Math.min(1, (wrongAnswers + correctAnswers) / Math.min(time / TIME_PER_QUESTION, numberOfQuestions))
```

In my opinion the `time` here should be the time the player needed to answer the questions.

Reply:
Because the chickenshock game will get changed so the lecturer can set the time per question, the calculate knowledge function will only devide correctAnswers by numberOfQuestions